### PR TITLE
Inference verification plot

### PR DESCRIPTION
**Pull request by clechartre**
## Purpose 

Implements an extra-feature of the vis module to perform standalone plotting of the predictions. This allows for the verification of the inference output which outputs a Numpy array (see[ PR#13](https://github.com/MeteoSwiss/neural-lam/pull/13)). Users can specify file path for saving and the channel (variable and level) to be plotted over the time range available in the inference output.  

## Code changes:

- `vis.py`: Addition of the standalone function `verify_inference`, described in the purpose section of this description

- `weather_dataset.py`: Addition of a split and specific datatype called "verification" which handles a numpy array instead of .zarr files, the current datatype supported by the model. The file is loaded as such, and, since it is backtransformed before storage, no specific transformation of the file is needed to pre-process it before plotting. 

- `cli_plotting.py`: Creation of a command line entry point to run the plotting on inference data as a standalone

- `README.md`: adapted the readme according to this new functionality

## Checklist

Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://meteoswiss.atlassian.net/wiki/spaces/UA/pages/20157433/Factory).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README)

## Review

For the review process follow the guidelines at [Checklist](https://meteoswiss.atlassian.net/wiki/spaces/UA/pages/20156488/Code+Review)

### CODE REVIEW

File: neural_lam/vis.py [hunk 1]
```diff
@@ -179,3 +181,90 @@ def plot_spatial_error(error, title=None, vrange=None):
...
+
+@matplotlib.rc_context(utils.fractional_plot_bundle(1))
+def verify_inference(file_path:str, feature_channel:int, vrange=None, save_path=None):
+    """
+    Plot example prediction, forecast, and ground truth.
+    Each has shape (N_grid,)
+    """
+
+    # Verify that feature channel is between 0 and 42
+    if not 0 <= feature_channel <= 42:
```
Comment (sadamov) [inline neural_lam/vis.py:193]: the number of feature channels is defined in constants.py by the user. 42 is only the current setting. This if clause should probably compare to the size of the last dimension of the .npy array from file_path. Or alternatively calculate them from `contants.py` param_names and is vertical lists.

File: neural_lam/vis.py [hunk 2]
```diff
@@ -179,3 +181,90 @@ def plot_spatial_error(error, title=None, vrange=None):
...
+    total =  predictions[0,:,:,feature_channel]
+    total_array = np.array(total)
+    if vrange is None:
+        vmin = total_array.min()
+        vmax = total_array.max()
+    else:
+        vmin, vmax = float(vrange[0].cpu().item()), float(vrange[1].cpu().item())
+
+    # Plot
+    for i in range(23):
```
Comment (sadamov) [inline neural_lam/vis.py:226]: Also here 23 is a value defined by the user in constants.py called `eval_horizon -2`. This loop should be more flexible and take the user input into account

File: neural_lam/weather_dataset.py [hunk 3]
```diff
@@ -43,63 +45,73 @@ def __init__(
             "val",
             "test",
             "pred",
+            "verification"
```
Comment (sadamov) [inline neural_lam/weather_dataset.py]: suggestion to rename this to "verif" everywhere to match the other splits

File: neural_lam/vis.py (hunk 1 above)
Comment (sadamov) [reply neural_lam/vis.py:193]: fixed in 7715d2be8001de4a2853cf448c8699efb29dee23

**Review by sadamov — COMMENTED** [review]
Thanks! This will be very useful when plotting/evaluating various input formats in a standalone script. The two comments wrt `23` and `42` should be adressed before merge.
I think this PR would benefit from an additional script (slurm, CLI) for the user to interact with the standalone plotting. Some guidance on how to use it should then be added to the README. Would you have time to add such functionality?

File: neural_lam/weather_dataset.py (hunk 3 above)
Comment (sadamov) [reply neural_lam/weather_dataset.py]: fixed in 7715d2be8001de4a2853cf448c8699efb29dee23

**Review by sadamov — APPROVED** [review]
Looks good! I made some smaller changes, as the code was not running on Balfrin initially. Please review my changes and if you agree squash commits and merge PR.
Thanks :pray:

**Comment by clechartre** [comment]
## Purpose 

Implements an extra-feature of the vis module to perform standalone plotting of the predictions. This allows for the verification of the inference output which outputs a Numpy array (see[ PR#13](https://github.com/MeteoSwiss/neural-lam/pull/13)). Users can specify file path for saving and the channel (variable and level) to be plotted over the time range available in the inference output.  


## Code changes:

- `vis.py`: Addition of the standalone function `verify_inference`, described in the purpose section of this description

- `weather_dataset.py`: Addition of a split and specific datatype called "verification" which handles a numpy array instead of .zarr files, the current datatype supported by the model. The file is loaded as such, and, since it is backtransformed before storage, no specific transformation of the file is needed to pre-process it before plotting. 

- `cli_plotting.py`: Creation of a command line entry point to run the plotting on inference data as a standalone

- `README.md`: adapted the readme according to this new functionality

**Pull request merged**